packages/samples/headless-react/src/components/category-facet/category-facet.class.tsx

```diff
@@ -1,34 +1,35 @@
-import {Component} from 'react';
+import {Component, ContextType} from 'react';
 import {
   buildCategoryFacet,
   CategoryFacet as HeadlessCategoryFacet,
   CategoryFacetState,
   CategoryFacetValue,
   Unsubscribe,
 } from '@coveo/headless';
-import {engine} from '../../engine';
 import {CategoryFacetSearch} from './category-facet-search';
+import {AppContext} from '../../context/engine';
 
 interface CategoryFacetProps {
   field: string;
   facetId: string;
 }
 
-export class CategoryFacet extends Component<CategoryFacetProps> {
-  private controller: HeadlessCategoryFacet;
-  public state: CategoryFacetState;
-  private unsubscribe: Unsubscribe = () => {};
+export class CategoryFacet extends Component<
+  CategoryFacetProps,
+  CategoryFacetState
+> {
+  static contextType = AppContext;
+  context!: ContextType<typeof AppContext>;
 
-  constructor(props: CategoryFacetProps) {
```

Comment:
Unfortunately, reading the context from the constructor is deprecated, which is what forces most of the changes in this PR: we're now forced to use lifecycle methods instead.